crowbar_framework/app/models/api/node.rb

```diff
@@ -15,6 +15,72 @@
 #
 
 module Api
-  class Node
+  class Node < Tableless
+    def repocheck(options = {})
+      addon = options.fetch(:addon, "os")
+      features = []
+      features.push(addon)
+      architectures = node_architectures
+
+      {}.tap do |ret|
+        ret[addon] = {
+          "available" => true
+        }
+        platform = Api::Upgrade.new.target_platform(platform_exception: addon)
+
+        features.each do |feature|
+          if architectures[feature]
+            architectures[feature].each do |architecture|
+              unless ::Crowbar::Repository.provided_and_enabled?(feature,
+                                                                 platform,
+                                                                 architecture)
+                ::Openstack::Upgrade.enable_repos_for_feature(feature, Rails.logger)
+              end
+              available, repolist = ::Crowbar::Repository.provided_and_enabled_with_repolist(
+                feature, platform, architecture
+              )
+              ret[addon]["available"] &&= available
+              ret[addon]["repos"] ||= {}
+              ret[addon]["repos"].deep_merge!(repolist.deep_stringify_keys)
+            end
+          else
+            ret.delete(addon)
+          end
+        end
+      end
+    end
+
+    protected
+
+    def node_architectures
+      {}.tap do |ret|
+        NodeObject.all.each do |node|
+          arch = node.architecture
+          ret["os"] ||= []
+          ret["os"].push(arch) unless ret["os"].include?(arch)
+
+          if ceph_node?(node)
+            ret["ceph"] ||= []
+            ret["ceph"].push(arch) unless ret["ceph"].include?(arch)
+          else
+            ret["openstack"] ||= []
+            ret["openstack"].push(arch) unless ret["openstack"].include?(arch)
+          end
+
+          if pacemaker_node?(node)
+            ret["ha"] ||= []
+            ret["ha"].push(arch) unless ret["ha"].include?(arch)
+          end
+        end
+      end
+    end
+
+    def ceph_node?(node)
+      node.roles.include?("ceph-mon")
```

Comment:
There might be different roles, not just `ceph-mon`, look into https://github.com/crowbar/crowbar-ceph/tree/master/chef/roles